### ISSUE TITLE
Issue for Different Game

### ISSUE THREAD

**Issue by jfw225** [ofeksadlo/CSGO-Aimbot-CNN#10]
I created a model for Escape from Tarkov and I am trying to write a similar library. I'm having a lot of trouble getting mouse movement to work in game, and in trying to find a solution, I stumbled upon this repo.

I've tried pyautogui, pydirectinput, SendInput (the same function in your repo) with no success. I see you say something about "raw input" needing to be off. Does this just mean turn off the "enhance pointer" thing, or is there something else I need to disable? If not, is there anything you ran into while developing that may be helpful?

Thanks.

**Comment by loggerMission** [comment]
Well, in CS:GO the raw input option will take the mouse input's directly.
Which in that case I cannot alter it.
So a way around it is to disable this option. Which exists in many First person shooters.

The enhance pointer is not relevant. It's just a mouse acceleration option
That for desktop is fine but for First person shooters is very bad.
Raw input will bypass this option but when we disable it.
We get this acceleration mouse movement in game.
So I manually disabled it.

Also you should try testing in full-screen windowed.
Good luck!

**Comment by ofeksadlo** [comment]
Well, in CS:GO the raw input option will take the mouse input's directly.
Which in that case I cannot alter it.
So a way around it is to disable this option. Which exists in many First person shooters.

The enhance pointer is not relevant. It's just a mouse acceleration option
That for desktop is fine but for First person shooters is very bad.
Raw input will bypass this option but when we disable it.
We get this acceleration mouse movement in game.
So I manually disabled it.

Also you should try testing in full-screen windowed.
Good luck!

**Comment by jfw225** [comment]
So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?

**Comment by ofeksadlo** [comment]
> So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?

I suppose yea.
Although there is no guarantee the option exists.

**Comment by jfw225** [comment]
Yeah sadly it doesn't. I don't think it's possible for tarkov with pure python or c libraries implemented in python, and I don't know enough C or about windows hooks to write my own library. I think I'm gonna try turning an Arduino into a mouse and then using python to write serial commands lol.

**Comment by ofeksadlo** [comment]
That's a bit of an over kill.
I mean it has to get the mouse input in some way.
You can research what game engine it uses and try to get started from there.

**Comment by maoshupoi** [comment]
> So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?

ETF's aiming mechanic is different from other games. It may have a value that can change the mouse movement, and this value is influenced by many things, such as the weight of the gun, the player's skill, etc.

**Comment by maoshupoi** [comment]
> So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?

I want to design something like trying to detect the line of iron sight and the target and put them in a line, like a real aiming.

**Comment by maoshupoi** [comment]
> > So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?
> 
> I want to design something like trying to detect the line of iron sight and the target and put them in a line, like a real aiming.

Maybe like this
![AIAiming](https://user-images.githubusercontent.com/75820866/145830271-ed95d399-eb47-40d8-a333-040440f8acda.png)
?

**Comment by ofeksadlo** [comment]
> > > So the "raw input" setting is a setting in CSGO, not Windows in general. So I should be looking for something similar in Tarkov?
> > 
> > 
> > I want to design something like trying to detect the line of iron sight and the target and put them in a line, like a real aiming.
> 
> Maybe like this ![AIAiming](https://user-images.githubusercontent.com/75820866/145830271-ed95d399-eb47-40d8-a333-040440f8acda.png) ?

Tha'ts the same principle my project is based on.
But in CS:GO the bullet will go were you shot. If it's more like BF (bullet won't go in direct line)
then it can get complicated but solvable.
You will need to get the distance between you and the player
according to how small the target is.

**Comment by maoshupoi** [comment]
> Tha'ts the same principle my project is based on. But in CS:GO the bullet will go were you shot. If it's more like BF (bullet won't go in direct line) then it can get complicated but solvable. You will need to get the distance between you and the player according to how small the target is.

I think there are the problem with Weapon shaking, like this one
![Weapon shaking](https://user-images.githubusercontent.com/75820866/145850654-cfea9583-98de-4a3d-9b78-ca46178f0d3b.gif)
The sight is not in the center of the screen.
